google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/RemoveFeedItemAttributeValue.java

```diff
@@ -0,0 +1,261 @@
+    int attributeId = (int) feedAttributes.get(flightPlaceholderField).getId().getValue();
+
+    // Retrieves the feed item and its associated attributes based on its resource name.
+    FeedItem feedItem = getFeedItem(googleAdsClient, customerId, feedItemResourceName);
+    // Creates the FeedItemAttributeValue that will be updated.
+    FeedItemAttributeValue feedItemAttributeValue =
+        FeedItemAttributeValue.newBuilder().setFeedAttributeId(Int64Value.of(attributeId)).build();
+    // Gets the index of the attribute value that will be removed.
+    int attributeIndex = getAttributeIndex(feedItem, feedItemAttributeValue);
+
+    // Returns the feed item with the removed FeedItemAttributeValue. Any FeedItemAttributeValues
+    // that are not included in the updated FeedItem will be removed from the FeedItem, which is
+    // why you must create the FeedItem from the existing FeedItem and set the field(s) that will
+    // be removed.
+//    return FeedItem.newBuilder(feedItem).removeAttributeValues(attributeIndex).build();
+    return feedItem.toBuilder().removeAttributeValues(attributeIndex).build();
+  }
+
+  /**
+   * Retrieves a feed item and its attribute values given a resource name.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID in which to create criterion.
+   * @param feedItemResourceName feed item resource name.
+   * @return FeedItem with the given resource name.
+   */
+  private FeedItem getFeedItem(
+      GoogleAdsClient googleAdsClient, long customerId, String feedItemResourceName) {
+    // Constructs the query.
+    String query =
+        "SELECT feed_item.attribute_values FROM feed_item WHERE feed_item.resource_name "
+            + "= '"
+            + feedItemResourceName
+            + "'";
+
+    // Constructs the request.
+    SearchGoogleAdsRequest request =
+        SearchGoogleAdsRequest.newBuilder()
+            .setCustomerId(String.valueOf(customerId))
+            .setPageSize(PAGE_SIZE)
+            .setQuery(query)
+            .build();
+
+    // Issues the search request.
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      SearchPagedResponse searchPagedResponse = googleAdsServiceClient.search(request);
+      // Returns the feed item attribute values.
+      return searchPagedResponse.getPage().getResponse().getResults(0).getFeedItem();
+    }
+  }
+
+  /**
+   * Gets the ID of the flight price attribute. This is needed to specify which
+   * FeedItemAttributeValue will be updated in the given FeedItem.
+   *
+   * @param feedItem the FeedItem that will be updated.
+   * @param newFeedItemAttributeValue the new FeedItemAttributeValue that will be updated.
+   * @return Int the index of the attribute.
+   */
+  private int getAttributeIndex(
+      FeedItem feedItem, FeedItemAttributeValue newFeedItemAttributeValue) {
+    int attributeIndex = -1;
```

Comment:
In Java it's more common to use null instead of a special value for cases like this. That'll require changing this to type `Integer`.